### ISSUE TITLE
Enhancement: [prefer-promise-reject-errors] Options to allow `any` and `unknown`

### ISSUE THREAD

**Issue by kripod** [typescript-eslint/typescript-eslint#10375]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-promise-reject-errors/

### Description

Similarly to the [options of `only-throw-error`](https://typescript-eslint.io/rules/only-throw-error/#options), the following should be available for `prefer-promise-reject-errors`, possibly turned on in an upcoming major version by default:

- `allowThrowingAny`
- `allowThrowingUnknown`

### Fail

```TypeScript
async function demo() {
  try {
    throw new Error();
  } catch (error: unknown) {
    /* eslint @typescript-eslint/prefer-promise-reject-errors: "error" */
    return await Promise.reject(error);
  }
}
```

### Pass

```TypeScript
async function demo() {
  try {
    throw new Error();
  } catch (error: unknown) {
    /* eslint @typescript-eslint/prefer-promise-reject-errors: ["error", { allowThrowingUnknown: true }] */
    return await Promise.reject(error);
  }
}
```

**Comment by kirkwaiblinger** [comment]
Agreed! I think it makes sense for the rules to have the same flexibility.

**Comment by bkks1004** [comment]
If possible, would you mind if I tried???

**Comment by injae-kim** [comment]
Fix PR: https://github.com/typescript-eslint/typescript-eslint/pull/10392
